client/containers/CommunityCreate/CommunityCreate.js

```diff
@@ -1,177 +1,144 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { useState } from 'react';
 import { Button, NonIdealState } from '@blueprintjs/core';
-import { ColorInput, GridWrapper, InputField, ImageUpload, PageWrapper } from 'components';
-import { hydrateWrapper, apiFetch, slugifyString } from 'utils';
+import { ColorInput, GridWrapper, InputField, ImageUpload } from 'components';
+import { apiFetch, slugifyString } from 'utils';
+import { usePageContext } from 'utils/hooks';
 
 require('./communityCreate.scss');
 
-const propTypes = {
-	communityData: PropTypes.object.isRequired,
-	loginData: PropTypes.object.isRequired,
-	locationData: PropTypes.object.isRequired,
-};
-
-class CommunityCreate extends Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			subdomain: '',
-			title: '',
-			description: '',
-			heroLogo: '',
-			accentColorDark: '#2D2E2F',
-			accentColorLight: '#FFFFFF',
-			createIsLoading: false,
-			createError: undefined,
-		};
-		this.onCreateSubmit = this.onCreateSubmit.bind(this);
-		this.onSubdomainChange = this.onSubdomainChange.bind(this);
-		this.onTitleChange = this.onTitleChange.bind(this);
-		this.onDescriptionChange = this.onDescriptionChange.bind(this);
-		this.onHeroHeaderLogoChange = this.onHeroHeaderLogoChange.bind(this);
-	}
+const CommunityCreate = () => {
```

Comment:
A future refactor like this could make use of the `<PageContext.Consumer>` component in the render function and skip the hooks conversion altogether. I'm happy to see our codebase gradually drop React class components, but I think it pays off to be deliberate about it — there have been several very subtle errors that have crept in when we've tried to hookify things on our way through other changes, especially when (as is not the case here) `useEffect` is involved.